specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/keys.json

```diff
@@ -1289,6 +1289,9 @@
           "description": "The bytes encoded as a base64url string."
         }
       },
+      "required": [
```

Comment:
Pure response models typically don't have required parameters. I understand from internal conversations this may be causing generator to also potentially return `undefined`, but I suggest you file a generator bug to avoid this for response models.

That said, I don't see a problem taking this. It's just not common in this scenario.

Reply:
@maorleger May I ask whether this is necessary for the JS SDK support?

Reply:
Thanks folks, appreciate the reviews! It's not strictly necessary, no. I also wasn't aware that it's not common (thanks @heaths!) - happy to close this PR and see if we can do something in the generator for JS, or in the convenience layer. I was just thinking that making the change here would ensure everyone gets it when generating.

But, definitely happy to defer to you if it's not the right thing to do - let me know

Reply:
Required does fill a need in response models. One reason that it is uncommon to specify it is that the openapi 2/swagger specification made some unfortunate decisions such stating that "`required` should not be used for `readOnly` properties" - which led to the assumption that `required` was meaningless when describing responses. This was fixed in openapi 3.

If the property (key/value) is guaranteed to be returned from the service, marking it as required makes perfect sense. If anything in the toolchain complains, please let me know.

Reply:
Thanks for the context / history.